tests/driver_lm75/main.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_lm75
+ *
+ * @{
+ * @file
+ * @brief       Test program for the driver for the LM75 temperature sensor.
+ *
+ * A test setup for the driver for the lm75 temperature sensor
+ * including support for the lm75a and tmp1075 sensors as well.
+ *
+ * @author      Vitor Batista <vitor.batista@ml-pa.com>
+ *
+ * @}
+ */
+
+#include "board.h"
+#include "lm75.h"
+#include "lm75_regs.h"
+#include "lm75_params.h"
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+#include "xtimer.h"
+#include "log.h"
+#include "kernel_defines.h"
+#include <stdio.h>
+#include <stdint.h>
+
+#define I2C_BUS       dev->lm75_params.i2c_bus
+#define I2C_ADDR      dev->lm75_params.i2c_addr
+
+/* prints currently set OS and HYST temperatures */
+int lm75_check_set_temperature_limits(lm75_t *dev) {
+
+    int t_os, t_hyst;
+    /* get already set OS and HYST values */
+    lm75_get_os_temp(dev, &t_os);
+    printf("Set OS temp is %d.%dºC\n",   t_os / dev->lm75_params.res->os_mult, \
+                                         t_os % dev->lm75_params.res->os_mult);
+
+    lm75_get_hyst_temp(dev, &t_hyst);
+    printf("Set HYST temp is %d.%dºC\n", t_hyst / dev->lm75_params.res->os_mult, \
+                                         t_hyst % dev->lm75_params.res->os_mult);
+    return 0;
+}
+
+/* This function prints the current temperature */
+int lm75_print_temperature(lm75_t *dev) {
+
+    int temp;
+    lm75_get_temperature(dev, &temp);
+
+    /* this is required not to discard the leading zero and
+    * output a false result in the case the temperature is x.0625 */
+    if (dev->lm75_params.res->temp_mult > 1000 \
+         && temp % dev->lm75_params.res->temp_mult == 625) {
```

Comment:
I'm confused. I though temp was in m°C? Why do we still need device specific parameters here?

Reply:
Because all variables used are integers, printing decimal values to the user must be done using the remainder operator. Also because reading the device temperature registers is done at the maximum resolution possible, in the TMP1075 sensor, in the event the temperature is something like x.0625ºC, the value of `temperature % 10000` would be just x.625, because the leftmost zero was discarded and a false value was printed to the user. Therefore, this was done so, in that single specific case, a zero will be added before the 625 to print the true value - which is x.0625.